src/Window.cpp

```diff
@@ -1432,33 +1695,205 @@ bool Window::moveForward(int distance) {
     int moves = 1;
     while (moves < distance) {
         if (wallFront(moves)) {
-            break;
+            stopRun("Stop running because of a wall!");
+            return false;
         }
         moves += 1;
     }
-    m_movement = Movement::MOVE_FORWARD;
+    m_movement = Movement::MOVE_FULLFORWARD;
     m_doomedToCrash = (moves != distance);
     m_movesRemaining = moves;
-    if (m_startingLocation.first == 0 && m_startingLocation.second == 0) {
+
+    if (m_startingLocation.first == 1 && m_startingLocation.second == 1) {
         stats->startRun();
     }
     // increase the stats by the distance that will be travelled
-    stats->addDistance(moves);
+
+    if(real)
```

Comment:
Just consult the status of the button here - you don't need the extra variable

Reply:
will do